### PR TITLE
Core/Spells: Fix monk roll

### LINKED ISSUE #18347: [master] Dash spells (ie: Roll, Chi Torpedo, Fel Dash) will not perform the movement
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

**Description:**
Various spells that move the player are not functioning as intended; missing scripts.

**Current behaviour:** (Tell us what happens.)
Using spells such as monk's roll and demon hunter's fel dash cause no player movement.

**Expected behaviour:** (Tell us what should happen instead.)
Fel dash dashes forward, and monk's roll rolls the player.

**Steps to reproduce the problem:**

1. Use spell (ie: roll)
2. See no player displacement
3. Cooldown begins/charge decays by one. 

**Branch(es):** 3.3.5 / master (Tell us which branch(es) this issue affects.) master

**TC rev. hash/commit:**  9b006e511bf40020e1660adf290ceb6ed6f4fd86

**TDB version:**  https://github.com/TrinityCore/TrinityCore/releases/tag/TDB703.00

**Operating system:**  Windows 8.1

[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)
[//]: # (For fixes containing C++, create a Pull Request.)

### PR DESCRIPTION
**Pull request by aquadeus**
**Changes proposed:**

-  Fix talent Roll

**Issues addressed:**

Closes #18347 Partially

**Tests performed:**

Tested in-game

### CODE REVIEW

File: src/server/scripts/Spells/spell_monk.cpp [hunk 1]
```diff
@@ -164,6 +166,36 @@ class spell_monk_provoke : public SpellScript
...
+    SpellCastResult CheckCast()
+    {
+        if (GetCaster()->HasUnitState(UNIT_STATE_ROOT))
+            return SPELL_FAILED_ROOTED;
+        return SPELL_CAST_OK;
+    }
+
+    void HandleDummy(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetCaster(), SPELL_MONK_ROLL_TRIGGER, true);
```
Comment (Ovahlord) [inline src/server/scripts/Spells/spell_monk.cpp:188]: does the sniff contain a smsg_spell_start for the triggered effects? If yes, remove the true

Comment (Ovahlord) [reply src/server/scripts/Spells/spell_monk.cpp:188]: ye, so no need to cast it as triggered = true then. Do normal casts instead. triggered = true skips the spell_start part coreside and goes straight for spell_go

Comment (aquadeus) [reply src/server/scripts/Spells/spell_monk.cpp:188]: oh ty so only GetCaster()->CastSpell(GetCaster(), SPELL_MONK_ROLL_TRIGGER) ? idk what is happening with this pr it doesnt detect changes or compiles XD

Comment (Shauren) [reply src/server/scripts/Spells/spell_monk.cpp:188]: I recommend triple checking that the script still works and spell casts properly before discarding that triggered argument

Comment (P-Kito) [reply src/server/scripts/Spells/spell_monk.cpp:188]: Have you tried rolling backwards? Just curious

**Pull request merged**